photon.c

```diff
@@ -0,0 +1,135 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <unistd.h>
+#include <sys/socket.h>
+#include <sys/un.h>
+#include <inttypes.h>
+
+#include "utarray.h"
+#include "common.h"
+#include "state/task_queue.h"
+#include "state/db.h"
+#include "event_loop.h"
+#include "task.h"
+#include "photon.h"
+#include "io.h"
+
+typedef struct {
+  event_loop *loop;
+  db_handle *db;
+  UT_array *task_queue;
+} local_scheduler_state;
+
+void init_local_scheduler(local_scheduler_state *s) {
+  s->loop = event_loop_create();
+};
+
+void process_message(event_loop *loop, int client_sock, void *context, int events) {
+  local_scheduler_state *s = context;
+
+  uint8_t *message;
+  int64_t type;
+  int64_t length;
+  read_message(client_sock, &type, &length, &message);
+
+  switch (type) {
+    case SUBMIT_TASK: {
+        task_spec *task = (task_spec *) message;
+        CHECK(task_size(task) == length);
+        unique_id id = globally_unique_id();
+        task_queue_submit_task(s->db, id, task);
+      }
+      break;
+    case TASK_DONE: {
+      }
+      break;
+    case DISCONNECT_CLIENT: {
+        LOG_INFO("Disconnecting client on fd %d", client_sock);
+        event_loop_remove_file(loop, client_sock);
+      }
+      break;
+    case LOG_MESSAGE: {
+      }
+      break;
+    default:
+      /* This code should be unreachable. */
+      CHECK(0);
+  }
+  free(message);
+}
+
+void new_client_connection(event_loop *loop, int listener_sock, void *context, int events) {
+  local_scheduler_state *s = context;
+  int new_socket = accept(listener_sock, NULL, NULL);
+  if (new_socket < 0) {
+    if (errno != EWOULDBLOCK) {
+      LOG_ERR("accept failed");
+      exit(-1);
+    }
```

Comment:
There is `accept_client` in `io.h`. Maybe we can move the errno check over there?